swap/swap_test.go

```diff
@@ -112,13 +113,29 @@ func newTestBackend() *swapTestBackend {
 	// commit the initial "pre-mined" accounts (issuer and beneficiary addresses)
 	defaultBackend.Commit()
 
+	tokenAddress, _, token, err := contractFactory.DeployERC20Mintable(bind.NewKeyedTransactor(ownerKey), defaultBackend)
+	if err != nil {
+		log.Crit(err.Error())
+	}
+	defaultBackend.Commit()
+
+	_, err = token.Mint(bind.NewKeyedTransactor(ownerKey), ownerAddress, big.NewInt(1000000000000000000))
+	if err != nil {
+		log.Crit(err.Error())
```

Comment:
? 

if this failed, we can just continue?

Reply:
It will be caught in the test using these tokens. I don't see how we could handle this more gracefully, as we don't have access to `t`, to call `t.Fatal()`, and we cannot return an error from this function.

Reply:
@ralph-pichler is the author of these lines, so please chip in if my answer is not complete

Reply:
I introduced that hack here because the helper function here does not have access to the testing object. During a test `log.Crit` automatically aborts and fails the test. Ideally though this should be refactored so that a `t.Fatal` can be used here.

Reply:
The easiest could be to just pass `t` to this function and call `t.Helper()`.

Reply:
@holisticode , just made the change you proposed